src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,22 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String producer;
+    private int countOfDoors;
+    private Engine engine;
+    private String color;
+    private String type;
 
+    public Car(String producer, int countOfDoors, Engine engine, String color, String type) {
+        this.producer = producer;
+        this.countOfDoors = countOfDoors;
+        this.engine = engine;
+        this.color = color;
+        this.type = type;
+    }
+
+    @Override
+    protected Object clone() {
+        return new Car(producer,countOfDoors,engine,color,type);
```

Comment:
```suggestion
        return new Car(producer, countOfDoors, engine.clone(), color, type);
```
we are interested in deep copy, with your impl you copying reference and not creating separate object